catalog/mvcc/cataloger_list_entries.go

```diff
@@ -283,17 +298,28 @@ func findLowestResultInBranches(branchRanges map[int64][]entryPathPrefixInfo, br
 	return chosenBranch
 }
 
-func buildBaseLevelQuery(baseBranchID int64, lineage []lineageCommit, branchEntryLimit int,
-	topCommitID CommitID, prefixLen int, endOfPrefixRange string) map[int64]sq.SelectBuilder {
+func buildBaseLevelQuery(baseBranchID int64, lineage []lineageCommit, branchBatchSize int,
+	requestedCommitID CommitID, prefixLen int, endOfPrefixRange string) map[int64]sq.SelectBuilder {
+	// builds a map of select queries for each of the branches in  the requested branch lineage
+	// number of entries that will be retrieved for each branch is limitted to branchBatchSize (The reason it is not enough
+	// to read a single row is that we may retrieve deleted entries or tombstones, that should be skipped.
+	// the requested commitID is passed to the base branch as is. each of the lineage branches gets the commit id from its lineage.
 	unionMap := make(map[int64]sq.SelectBuilder)
-	unionMap[baseBranchID] = buildSingleBranchQuery(baseBranchID, branchEntryLimit, topCommitID, prefixLen, endOfPrefixRange)
+	unionMap[baseBranchID] = buildSingleBranchQuery(baseBranchID, branchBatchSize, requestedCommitID, prefixLen, endOfPrefixRange)
 	for _, l := range lineage {
-		unionMap[l.BranchID] = buildSingleBranchQuery(l.BranchID, branchEntryLimit, l.CommitID, prefixLen, endOfPrefixRange)
+		unionMap[l.BranchID] = buildSingleBranchQuery(l.BranchID, branchBatchSize, l.CommitID, prefixLen, endOfPrefixRange)
 	}
 	return unionMap
 }
 
 func buildSingleBranchQuery(branchID int64, branchBatchSize int, topCommitID CommitID, prefixLen int, endOfPrefixRange string) sq.SelectBuilder {
+	// builds a query on a single branch of the lineage. called mainly from "buildBaseLevelQuery" above.
+	// the other function that calls it is "getMoreRows" that needs entries for a single branch.
+	// topCommitId contains the requested commit for that branch. its implications:
+	// 1. entries where the minCommitId is more than the requested commit id will be filtered out
+	// 2. entries that were deleted after this commit (maxCommitId > topCommitId) will be considered
```

Comment:
This reads a bit like describing how the feature was implemented more than what the parameter does.  I _think_ that an equivalent description (with fewer details of implementation, and instead much more focus on what it does when you pass it):
> Returns entries as they were at topCommitID.
(Of course if this is wrong then don't copy this wrong description :smiley_cat: )